### PR TITLE
Update inliner to propagate valueinfos

### PR DESCRIPTION
**Pull request by gramalingam**
### Description

This is a follow up to the PR adding valueinfos field to FunctionProto. This PR updates the inliner to propagate this information from the inlined function.

### CODE REVIEW

File: onnx/inliner/inliner.h
Comment (BowenBao) [inline onnx/inliner/inliner.h:1]: I recall the onnx inliner python api accepts list of FunctionIds, is there plan to update it?

File: onnx/inliner/inliner.cc [hunk 1]
```diff
@@ -283,6 +267,8 @@ class InliningRenamer : private MutableVisitor {
     renamer.Bind<true>(*callee.mutable_output(), callnode.output());
 
     renamer.VisitFunction(&callee);
+    for (auto& v : *callee.mutable_value_info())
+      renamer.LookupOrRename(*v.mutable_name(), false);
```
Comment (BowenBao) [inline onnx/inliner/inliner.cc:271]: what happens for function inputs and outputs? Will they be renamed/bind to callnode inputs and outputs?

File: onnx/inliner/inliner.cc [hunk 2]
```diff
@@ -492,6 +480,9 @@ void InlineFunctions(
             callee.name().c_str());
         ConvertVersion(*model, node, callee, target_version);
       }
+      // Append valueinfos of called function
+      for (auto& callee_vi : callee.value_info())
+        *value_infos.Add() = callee_vi;
```
Comment (BowenBao) [inline onnx/inliner/inliner.cc:485]: Follow up on https://github.com/onnx/onnx/pull/5942/files#r1498435308, will this append value_info of function inputs outputs?

File: onnx/inliner/inliner.h
Comment (BowenBao) [reply onnx/inliner/inliner.h:1]: Also curious if there are any python api unittests that we should update

File: onnx/inliner/inliner.cc (hunk 2 above)
Comment (gramalingam) [reply onnx/inliner/inliner.cc:485]: Good point. Maybe we should not insert duplicates (which can happen only for inputs/outputs).

File: onnx/inliner/inliner.h
Comment (gramalingam) [reply onnx/inliner/inliner.h:1]: Yes, I was wondering about this (whether to break backward compatibility or not). Decided on what the comments say below. I think we can maintain compatibility for existing API. We can separately add new APIs if user wants to inline overload1 of a function but not overload2 of the same function.

File: onnx/inliner/inliner.cc (hunk 2 above)
Comment (gramalingam) [reply onnx/inliner/inliner.cc]: Done

**Review by BowenBao — APPROVED** [review]

**Review by liqunfu — APPROVED** [review]

**Review by thiagocrepaldi — APPROVED** [review]

**Pull request merged**